python/hail/table.py

```diff
@@ -1283,17 +1282,29 @@ def types_compatible(left, right):
             for e in exprs:
                 analyze('Table.index', e, src._row_indices)
 
-            right = self.select(**{uid: self.row})
-            uids = [Env.get_uid() for i in range(len(exprs))]
+            is_key = src.key is not None and len(exprs) == len(src.key) and all(
+                exprs[i] is src._fields[list(src.key)[i]] for i in range(len(exprs)))
```

Comment:
is this more clear as `... for (i, expr) in enumerate(exprs)`?

Reply:
definitely